AEPTarget/Sources/Target.swift

```diff
@@ -845,30 +869,90 @@ public class Target: NSObject, Extension {
         MobileCore.dispatch(event: event)
     }
 
-    /// Grabs the a4t payload from the target response and convert the keys to correct format
-    /// Returns an empty `Dictionary` if there are no analytics payload that needs to be sent.
+    /// Preprocesses the analytics for target (a4t) payload received in the Target response for Analytics consumption.
+    /// - Returns: `Dictionary` containing a4t payload with keys in internal format.
+    private func preprocessAnalyticsPayload(_ payload: [String: String], sessionId: String) -> [String: String] {
+        var result: [String: String] = [:]
+
+        for (k, v) in payload {
+            result["&&\(k)"] = v
+        }
+
+        if !sessionId.isEmpty {
+            result[TargetConstants.TargetJson.SESSION_ID] = sessionId
+        }
+
+        return result
+    }
+
+    /// Grabs the analytics for target (a4t) payload from the target response if available.
     /// - Parameters:
-    ///     - mboxJson: A prefetched mbox dictionary
-    ///     - sessionId: session id
-    /// - Returns: `Dictionary` containing a4t payload
-    private func getAnalyticsForTargetPayload(mboxJson: [String: Any], sessionId: String) -> [String: String]? {
-        var payload: [String: String] = [:]
-        guard let analyticsJson = mboxJson[TargetConstants.TargetJson.ANALYTICS_PARAMETERS] as? [String: Any] else {
+    ///     - json: dictionary containing a4t payload.
+    /// - Returns: `Dictionary` containing a4t payload or nil.
+    private func getAnalyticsForTargetPayload(json: [String: Any]) -> [String: String]? {
+        guard let analyticsJson = json[TargetConstants.TargetJson.ANALYTICS] as? [String: Any] else {
             return nil
         }
 
         guard let payloadJson = analyticsJson[TargetConstants.TargetJson.ANALYTICS_PAYLOAD] as? [String: String] else {
             return nil
         }
 
-        for (k, v) in payloadJson {
-            payload["&&\(k)"] = v
+        return payloadJson
+    }
+
+    /// Extracts click metric info from the Target response if available.
+    /// - Parameters:
+    ///     - mboxJson: Mbox dictionary.
+    /// - Returns: tuple containing `String` click token and `Dictionary` containing click tracking analytics payload.
+    private func extractClickMetric(mboxJson: [String: Any]) -> (token: String?, analyticsPayload: [String: String]?) {
+        guard let metrics = mboxJson[TargetConstants.TargetJson.METRICS] as? [[String: Any]?] else {
+            return (nil, nil)
         }
 
-        if !sessionId.isEmpty {
-            payload[TargetConstants.TargetJson.SESSION_ID] = sessionId
+        for metricItem in metrics {
+            guard
+                let metric = metricItem,
+                TargetConstants.TargetJson.MetricType.CLICK == metric[TargetConstants.TargetJson.Metric.TYPE] as? String,
+                let token = metric[TargetConstants.TargetJson.Metric.EVENT_TOKEN] as? String,
+                !token.isEmpty
```

Comment:
If the token is empty and mbox click has valid a4t params. Shouldn't  we return them to client?

Reply:
Nope. You do not want to send a4t data if the click cannot be registered in Target in the first place. Ideally, we shouldn't run into such a case but it's good to check just in case.